ctapipe/io/hessiofilereader.py

```diff
@@ -0,0 +1,211 @@
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.core import Provenance
+from ctapipe.io.eventfilereader import EventFileReader
+from ctapipe.io.containers import DataContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription
+
+
+COUNT = 0
+
+
+class HessioFileReader(EventFileReader):
+
+    def __init__(self, config, tool, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        try:
+            from pyhessio import open_hessio
+            from pyhessio import close_file
+            from pyhessio import HessioError
+            from pyhessio import HessioTelescopeIndexError
+            from pyhessio import HessioGeneralError
+        except ImportError:
+            msg = "The `pyhessio` python module is required to access MC data"
+            self.log.error(msg)
+            raise
+
+        self.open_hessio = open_hessio
```

Comment:
I do not understand why we gain from assigning these functions and Exceptions to the reader, this looks like just adding noise.

Reply:
Okay, I have removed the unused ones. I need to keep a reference to `open_hessio` and `close_file` for other methods in the class.

Reply:
Why? Why is the top level import not enough?

Reply:
We will be directing EventFileReader-creator to perform their import of external modules in the `__init__` method to not introduce dependencies on those modules.

To then be able to use that module elsewhere in the class, one must keep a reference to the import.

As pyhessio is a dependency of ctapipe, in this case we could import pyhessio at the top of this module instead of inside `__init__`, but I thought it better to keep things consistent between EventFileReaders, and besides maybe one day we will drop the dependency on pyhessio.

Reply:
I have reduced clutter now by only keeping a reference to the pyhessio module itself, and not its separate functions